### ISSUE TITLE
Kokkos CMake should find CUDA distributed with NVHPC SDK

### ISSUE THREAD

**Issue by jeffhammond** [kokkos/kokkos#4191]
The [NVIDIA HPC SDK](https://developer.nvidia.com/nvidia-hpc-sdk-downloads) comes with CUDA included.

CMake in Kokkos cannot find the `libcudart.so` from this distribution even though it finds the `nvcc`.

Please support NVHPC with its included CUDA.

Given that the NVHPC SDK can be installed in user space with simple commands (`wget`/`curl`+`tar`+shell), it would be easy to add it to your CI system (for x86, Power and AArch64, if you so desire).

Thanks @DavidPoliakoff 

```
~/kokkos/build$ ../generate_makefile.bash --with-cuda 
reseting kokkos devices to Cuda
parsing KOKKOS_OPTIONS=
parsing KOKKOS_CUDA_OPTIONS=
parsing KOKKOS_HIP_OPTIONS=
parsing KOKKOS_OMPT_OPTIONS=
cmake -DCMAKE_CXX_FLAGS= -DCMAKE_EXE_LINKER_FLAGS= -DCMAKE_INSTALL_PREFIX= -DKokkos_ENABLE_CUDA=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_BUILD_TYPE=RELEASE ..
-- Setting default Kokkos CXX standard to 14
-- The CXX compiler identification is GNU 9.3.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting policy CMP0074 to use <Package>_ROOT variables
-- The project name is: Kokkos
-- Compiler Version: 11.2.152
-- kokkos_launch_compiler (/home/jehammond/PRK/gpu-gearbox/kokkos/bin/kokkos_launch_compiler) is enabled...
-- SERIAL backend is being turned on to ensure there is at least one Host space. To change this, you must enable another host execution space and configure with -DKokkos_ENABLE_SERIAL=OFF or change CMakeCache.txt
-- Using -std=c++14 for C++14 standard as feature
-- CUDA auto-detection of architecture failed with /usr/bin/c++. Enabling CUDA language ONLY to auto-detect architecture...
-- Looking for a CUDA compiler
-- Looking for a CUDA compiler - /opt/nvidia/hpc_sdk/Linux_x86_64/21.5/compilers/bin/nvcc
-- The CUDA compiler identification is NVIDIA 11.2.152
-- Check for working CUDA compiler: /opt/nvidia/hpc_sdk/Linux_x86_64/21.5/compilers/bin/nvcc
-- Check for working CUDA compiler: /opt/nvidia/hpc_sdk/Linux_x86_64/21.5/compilers/bin/nvcc -- works
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Detected CUDA Compute Capability 80
-- Setting Kokkos_ARCH_AMPERE80=ON
-- Built-in Execution Spaces:
--     Device Parallel: Kokkos::Cuda
--     Host Parallel: NoTypeDefined
--       Host Serial: SERIAL
-- 
-- Architectures:
--  AMPERE80
-- Unable to find cudart library.
-- Could NOT find CUDAToolkit (missing: CUDAToolkit_INCLUDE_DIR CUDA_CUDART) (found version "11.2.152")
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find TPLCUDA (missing: FOUND_CUDART FOUND_CUDA_DRIVER)
```

**Comment by robertmaynard** [comment]
Previously to CMake 3.20 the location of the cuda libraries was computed based on assumptions of the CUDA Toolkit layout.

For CMake 3.20 we had multiple changes that overall improved NVHPC support, the robustness of CUDA language and FindCUDAToolkit ( support for `nvcc` symlinks, and splayed directory layouts that NVHPC uses )

In short, users that are currently using NVHPC will need CMake 3.20.1+ . Going forward Kokkos should look at either backporting newer versions of the FindCUDAToolkit or bumping the minimum required CMake version for CUDA builds.

**Comment by dalg24** [comment]
We'd need to update
https://github.com/kokkos/kokkos/blob/develop/cmake/Modules/CudaToolkit.cmake
and https://github.com/kokkos/kokkos/blob/4b97a22ff7be7635116930bb97173058d6079202/cmake/Modules/FindTPLCUDA.cmake#L10-L14

**Comment by DavidPoliakoff** [comment]
Vote: in favor of requiring 3.20 to use nvhpc

SF F N A SA
3 9 1 0 0

Strong consensus to require 3.20

**Comment by PhilMiller** [comment]
Fix released in 3.6
